File: frontend/src/models/book.ts
import { Review } from "./review";

export interface Book {
    id: number;
    title: string;
    author: string;
    publicationYear?: number | null; // matches GetBookDTO.publicationYear
    isbn?: string | null;
    description?: string | null;
    reviews?: Review[];  // matches GetBookDTO.reviews
  }
  
  export interface BookPageDTO {
    content: Book[];
    totalElements: number;
    totalPages: number;
    currentPage: number;
  }

File: book.manager/src/main/java/strayfurther/book/manager/model/Book.java
package strayfurther.book.manager.model;

import jakarta.persistence.*;
import lombok.*;

import java.util.ArrayList;
import java.util.List;

@Entity
@Getter
@Setter
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Table(name="books")
public class Book {

	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	private Long id;

	@Column(nullable = false, length = 255)
	private String title;

	@Column(nullable = false, length = 255)
	private String author;

	@Column(length = 13)
	private String isbn;

	@Column(name = "publication_year")
	private Integer publicationYear;

	@Column(columnDefinition = "TEXT")
	private String description;

	@OneToMany(mappedBy = "book", cascade = CascadeType.REMOVE, fetch = FetchType.LAZY)
	private List<Review> reviews = new ArrayList<>();
}
